doc/acl/SONiC_ACL_based_metering_hld.md

```diff
@@ -0,0 +1,430 @@
+                                    type string;
+                                    default "";
+                            }
+                    }
+            }
+            container ACL_TABLE {
+
+                    description "ACL_TABLE part of config_db.json";
+
+                    list ACL_TABLE_LIST {
+
+                            key "ACL_TABLE_NAME";
+                            :
+                            :
++                           leaf-list actions {
++                                   type enumeration {
++                                       enum PACKET_ACTION;
++                                       enum POLICER;
++                                       enum REDIRECT;
++                                   }
++                           }
+                    }
+            }
+    }
+}
+
+```
+
+### Warmboot and Fastboot Considerations
+
+No impact to warm or fastboot
+
+### Assumptions/Limitations
+Since SAI has not defined the precedence of packet command derived between ACL RULE's packet_action and ACL RULE's policer action, it is assumed that packet-action of policer object takes precedence over ACL rule packet-action field. For example, if both are set like below, packet-action of policer should be applied by SAI. In this case, it is vendor SAI responsibility to apply this behavior and NOS will set both SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION and SAI_ACL_ENTRY_ATTR_ACTION_SET_POLICER to vendor SAI. If policer configuration is removed from ACL, SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION should be applied to the final packet command by SAI.
+
+In below example, aclorch will set sai acl entry action as SAI_ACL_ACTION_TYPE_PACKET_ACTION, SAI_ACL_ACTION_TYPE_SET_POLICER.
+
+```
+"policer2": {
+		    "meter_type": "bytes",
+		    "mode": "sr_tcm",
+		    "cir": "12500",
+		    "cbs": "12500",
+		    "pir": "17500",
+		    "pbs": "17500",
+		    "color": "aware",
+		    "red_packet_action": "drop",
+		    "yellow_packet_action": "drop",
+		    "green_packet_action": "trap"
+        }
+
+ "VRF_ACL_REDIRECT_V6|rule1": {
+            "priority": "55",
+            "SRC_IPV6": "2000:0030::1",
+            "packet_action": "FORWARD",
+            "policer_action":"policer2"
+        }
+```
+If policer action set across rules from different table members of a table-group, the final resolution should be decided based on "SAI_ACL_TABLE_GROUP_MEMBER_ATTR_PRIORITY".
+
+SAI documentation is not clear about the behavior when different policers are attached in each stage for the same traffic flow and assumption here is, if policers are attached at different stages in pipeline, traffic should be policed in each of the stage. Say,
+- storm control policer P1 attached to Ingress port port_1
+- poicre P2 attached to Ingress acl matching flow "x", bind to port_1
+
+Say for flow "x", P1 is applied first, and the remaining traffic is policed by P2. Other traffic flows are applied with policer P1.
+### Testing Design
+
+#### VS Test cases
```

Comment:
Please plan to have unit tests in swss instead of python VS tests.

Reply:
Sure. Going forward, is python VS test is optional ?

Reply:
Not exactly. There would be cases which cannot be covered by unit-test in which case we can use VS test.